molecular/src/test/kotlin/com/hartwig/actin/molecular/orange/MolecularRecordAnnotatorTest.kt

```diff
@@ -30,4 +61,54 @@ class MolecularRecordAnnotatorTest {
             assertThat(tumorMutationalLoad).isNull()
         }
     }
+
+    @Test
+    fun `Should annotate variant that is hotspot`() {
+        val evidenceDatabase = mockk<EvidenceDatabase> {
+            every { alterationForVariant(VARIANT_MATCH_CRITERIA.copy(driverLikelihood = null)) } returns HOTSPOT
+            every { evidenceForVariant(any()) } returns EMPTY_MATCH
+        }
+
+        val annotated = MolecularRecordAnnotator(evidenceDatabase).annotateVariant(VARIANT)
+        assertThat(annotated.isHotspot).isTrue()
+        assertThat(annotated.driverLikelihood).isEqualTo(DriverLikelihood.HIGH)
+        assertThat(annotated.proteinEffect.name).isEqualTo(HOTSPOT.proteinEffect.name)
+        assertThat(annotated.geneRole.name).isEqualTo(HOTSPOT.geneRole.name)
+    }
+
+    @Test
+    fun `Should annotate variant that is no hotspot`() {
+        val evidenceDatabase = mockk<EvidenceDatabase> {
+            every { alterationForVariant(VARIANT_MATCH_CRITERIA.copy(driverLikelihood = null)) } returns NON_HOTSPOT
+            every { evidenceForVariant(any()) } returns EMPTY_MATCH
+        }
+
+        val annotated = MolecularRecordAnnotator(evidenceDatabase).annotateVariant(VARIANT)
+        assertThat(annotated.isHotspot).isFalse()
+        assertThat(annotated.driverLikelihood).isEqualTo(VARIANT.driverLikelihood)
+        assertThat(annotated.proteinEffect.name).isEqualTo(NON_HOTSPOT.proteinEffect.name)
+        assertThat(annotated.geneRole.name).isEqualTo(NON_HOTSPOT.geneRole.name)
+    }
+
+    @Test
+    fun `Should reannotate driver likelihood on gene level`() {
+        val variants = listOf(
+            TestVariantFactory.createMinimal().copy(driverLikelihood = DriverLikelihood.HIGH),
+            TestVariantFactory.createMinimal().copy(driverLikelihood = DriverLikelihood.MEDIUM),
+            TestVariantFactory.createMinimal().copy(driverLikelihood = DriverLikelihood.LOW),
+        )
+        val output = annotator.reannotateDriverLikelihood(variants)
+        assertThat(output.all { it.driverLikelihood == DriverLikelihood.HIGH }).isTrue()
+    }
+
+    @Test
+    fun `Should not reannotate driver likelihood when variant has driverlikelihood null`() {
+        val variants = listOf(
+            TestVariantFactory.createMinimal().copy(driverLikelihood = DriverLikelihood.HIGH),
+            TestVariantFactory.createMinimal()
+        )
+        val output = annotator.reannotateDriverLikelihood(variants)
+        assertThat(output[0].driverLikelihood).isEqualTo(DriverLikelihood.HIGH)
+        assertThat(output[1].driverLikelihood).isNull()
+    }
```

Comment:
Two suggestions:
 - I would explicitly set the gene for every variant since your logic depends on having identical genes
 - I would add a test that driver likelihood is not updated for variants on different genes. In theory your code could now update the DL to HIGH for _all_ variants found, there is no test that safeguards for that!